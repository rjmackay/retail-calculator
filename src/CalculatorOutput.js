import React, { Component } from 'react';

class CalculatorOutput extends Component {
  render() {
    let total = getTotal(this.props);
    let taxRate = getTaxRate(this.props.stateCode);
    let tax = getTax(total, taxRate);
    return (
      <div className="CalculatorOutput">
        <h3>Results:</h3>
        <p>
          Number of items: {this.props.numberOfItems}<br />
          Price per item: ${this.props.pricePerItem}
        </p>
        <p>
          Subtotal: ${total ? total : ''}<br />
          Tax: ${tax ? tax : ''} ({taxRate}%)<br />
          Total: ${total ? total : ''} (excl. tax)<br />
        </p>
      </div>
    );
  }
}

export const getTotal = (submission) => {
  return submission.numberOfItems * submission.pricePerItem;
};

const stateCodeMap = {
  UT: 6.85,
  TX: 6.25,
  CA: 8.25,
  NV: 8.0,
  AL: 4.0
}

export const getTaxRate = (stateCode) => {
  return stateCodeMap[stateCode];
}

export const getTax = (subtotal, taxRate) => {
  return (subtotal / 100) * taxRate;
}

export default CalculatorOutput;
