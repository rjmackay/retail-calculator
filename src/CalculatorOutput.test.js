import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CalculatorOutput, { getTotal, getTaxRate } from './CalculatorOutput';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CalculatorOutput />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('', () => {
  const component = renderer.create(
    <CalculatorOutput numberOfItems="100" pricePerItem="200" />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('getTotal calculates total', () => {
  expect(getTotal({
    numberOfItems: 100,
    pricePerItem: 300
  })).toEqual(30000);
});

it('getTaxRate returns correct rates', () => {
  expect(getTaxRate('UT')).toEqual(6.85);
  expect(getTaxRate('TX')).toEqual(6.25);
  expect(getTaxRate('CA')).toEqual(8.25);
  expect(getTaxRate('NV')).toEqual(8.0);
  expect(getTaxRate('AL')).toEqual(4.0);
});
