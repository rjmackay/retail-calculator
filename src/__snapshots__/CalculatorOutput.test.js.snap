// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="CalculatorOutput"
>
  <h3>
    Results:
  </h3>
  <p>
    Number of items: 
    100
    <br />
    Price per item: $
    200
  </p>
  <p>
    Subtotal: $
    20000
    <br />
    Tax: $
    1370
     (
    6.85
    %)
    <br />
    Total: $
    20000
     (excl. tax)
    <br />
  </p>
</div>
`;
